### PR TITLE
tools/dm: add doc for DM use in shard merge scenario

### PR DESCRIPTION
**Pull request by dcalvin**
Added doc for DM use in shard merge scenario. The corresponding source file in English is:https://github.com/pingcap/docs/blob/master/tools/dm/shard-merge-scenario.md

@lilin90 @GregoryIan  PTAL

There are 4 external links included in this file, which are to be consistent with **data-synchronization-features.md** to be done by @GregoryIan 

/tools/dm/data-synchronization-features.md##table-routing
/tools/dm/data-synchronization-features.md#binlog-event-filtering
/tools/dm/data-synchronization-features.md#black-and-white-table-lists
/tools/dm/data-synchronization-features.md#column-mapping

### CODE REVIEW

File: tools/dm/shard-merge-scenario.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+## 同步需求
+
+1. 合并三个上游实例中的 `user`.`information` 表至下游 TiDB 中的 `user`.`information` 表。
+2. 合并三个上游实例中的 `user`.`log_{north|south|east}` 表至下游TiDB中的 `user`.`log_{north|south|east}` 表。
+3. 合并三个上游实例中的 `store_{01|02}`.`sale_{01|02}` 表至下游TiDB中的 `store`.`sale` 表。
+4. 从三个上游实例中的 `user`.`log_{north|south|east}` 表中过滤掉所有删除操作。
+5. 从三个上游实例中的 `user`.`information` 表中过滤掉所有删除操作。
+6. 从三个上游实例中的  `store_{01|02}`.`sale_{01|02}` 表中过滤掉所有删除操作。
+7. 过滤掉三个上游实例中的 `user`.`log_bak` 表。
+8. 因为 `store_{01|02}`.`sale_{01|02}` 表带有长整型的自增主键，将这些表合并至 TiDB 时会引发合并冲突。您需要修改相应自增主键以避免冲突。
```
Comment (csuzhangxc) [inline tools/dm/shard-merge-scenario.md]: `bigint` is better than `长整型` here.

File: tools/dm/shard-merge-scenario.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    | user  | information, log_south, log_bak |
+    | store_01 | sale_01, sale_02 |
+    | store_02 | sale_01, sale_02 |
+
+## 同步需求
+
+1. 合并三个上游实例中的 `user`.`information` 表至下游 TiDB 中的 `user`.`information` 表。
+2. 合并三个上游实例中的 `user`.`log_{north|south|east}` 表至下游TiDB中的 `user`.`log_{north|south|east}` 表。
+3. 合并三个上游实例中的 `store_{01|02}`.`sale_{01|02}` 表至下游TiDB中的 `store`.`sale` 表。
+4. 从三个上游实例中的 `user`.`log_{north|south|east}` 表中过滤掉所有删除操作。
```
Comment (csuzhangxc) [inline tools/dm/shard-merge-scenario.md]: I think
```
过滤掉三个实例的 `user``.log_{north|south|east}` 表的所有删除操作
```
in the original CN doc( <https://raw.githubusercontent.com/pingcap/tidb-tools/docs/docs/dm/zh_CN/use-cases/shard-merge.md>) is better, same bellow.

File: tools/dm/shard-merge-scenario.md [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+    ```yaml
+    routes:
+      ...
+      user-route-rule:
+        schema-pattern: "user"
+        target-schema: "user"
+    ```
+
+- 要满足同步要求 #3, 配置 [表路由规则](/tools/dm/data-synchronization-features.md##table-routing) 如下：
```
Comment (csuzhangxc) [inline tools/dm/shard-merge-scenario.md]: unify `同步要求` to `同步需求`?

File: tools/dm/shard-merge-scenario.md [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+    ```yaml
+    filters:
+      ...
+      user-filter-rule:
+        schema-pattern: "user"
+        events: ["truncate table", "drop table", "delete"， "drop database"]
+        action: Ignore
+    ```
+
+    > **注意：** 同步要求 #4、#5 和 #7 中，所有对 `user` 库的删除操作都要被过滤，所以此处配置了库级别的过滤规则。但需要注意的是，`user` 库将来出现表的删除操作也都会被过滤。
```
Comment (csuzhangxc) [inline tools/dm/shard-merge-scenario.md]: I think the original CN doc (<https://github.com/pingcap/tidb-tools/blob/docs/docs/dm/zh_CN/use-cases/shard-merge.md>) for this line is better.

File: tools/dm/shard-merge-scenario.md [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+假设同步后下游库结构如下：
+
+| 库 | 表 |
+|:------|:------|
+| user | information, log_north, log_east, log_south|
+| store | sale |
+
+## 同步方案
+
+- 要满足同步需求 #1 和 #2, 配置 [表路由规则](/tools/dm/data-synchronization-features.md##table-routing) 如下：
```
Comment (amyangfei) [inline tools/dm/shard-merge-scenario.md]:
```suggestion
- 要满足同步需求 #1 和 #2, 配置 [表路由规则](/tools/dm/data-synchronization-features.md#table-routing) 如下：
```

File: tools/dm/shard-merge-scenario.md (hunk 3 above)
Comment (amyangfei) [reply tools/dm/shard-merge-scenario.md]: link anchor should be `/tools/dm/data-synchronization-features.md#table-routing`

File: tools/dm/shard-merge-scenario.md (hunk 1 above)
Comment (dcalvin) [reply tools/dm/shard-merge-scenario.md]: got it.

File: tools/dm/shard-merge-scenario.md (hunk 2 above)
Comment (dcalvin) [reply tools/dm/shard-merge-scenario.md]: Indeed. Updated.

**Comment by csuzhangxc** [comment]
LGTM

File: tools/dm/shard-merge-scenario.md [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+## 同步需求
+
+1. 合并三个实例中的 `user`.`information` 表至下游 TiDB 中的 `user`.`information` 表。
+2. 合并三个实例中的 `user`.`log_{north|south|east}` 表至下游TiDB中的 `user`.`log_{north|south|east}` 表。
+3. 合并三个实例中的 `store_{01|02}`.`sale_{01|02}` 表至下游TiDB中的 `store`.`sale` 表。
+4. 过滤掉三个实例的 `user`.`log_{north|south|east}` 表的所有删除操作。
+5. 过滤掉三个实例的 `user`.`information` 表的所有删除操作。
+6. 过滤掉三个实例的 `store_{01|02}`.`sale_{01|02}` 表的所有删除操作
+7. 过滤掉三个实例的 `user`.`log_bak` 表。
+8. 因为 `store_{01|02}`.`sale_{01|02}` 表带有 bigint 型的自增主键，将其合并至 TiDB 时会引发冲突。您需要修改相应自增主键以避免冲突。
```
Comment (GregoryIan) [inline tools/dm/shard-merge-scenario.md]:
```suggestion
8. 因为 `store_{01|02}`.`sale_{01|02}` 表带有 bigint 型的自增主键，将其合并至 TiDB 时会引发冲突。您需要有方案修改相应自增主键以避免冲突。
```

**Comment by GregoryIan** [comment]
LGTM

File: tools/dm/shard-merge-scenario.md [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+
+本文介绍如何在分片合并场景中使用 Data Migration (DM)。使用场景中，三个上游 MySQL 实例的分库和分表数据需要同步至下游 TiDB 集群。
+
+## 上游实例
+
+假设上游库结构如下：
+
+- 实例 1
+
+    | 库 | 表 |
```
Comment (lilin90) [inline tools/dm/shard-merge-scenario.md]:
```suggestion
    | Schema | Tables |
```

Comment (lilin90) [reply tools/dm/shard-merge-scenario.md]: We just keep the English version here. Please also update other similar terms in this PR. Thanks!

File: tools/dm/shard-merge-scenario.md [hunk 8]
```diff
@@ -0,0 +1,273 @@
+---
+title: DM 分片合并场景
```
Comment (lilin90) [inline tools/dm/shard-merge-scenario.md]:
```suggestion
title: DM 分库分表合并场景
```

Comment (lilin90) [reply tools/dm/shard-merge-scenario.md]: Please also update similar expressions in this PR.

File: tools/dm/shard-merge-scenario.md [hunk 9]
```diff
@@ -0,0 +1,273 @@
...
+    | 库 | 表 |
+    |:------|:------|
+    | user  | information, log_south, log_bak |
+    | store_01 | sale_01, sale_02 |
+    | store_02 | sale_01, sale_02 |
+
+## 同步需求
+
+1. 合并三个实例中的 `user`.`information` 表至下游 TiDB 中的 `user`.`information` 表。
+2. 合并三个实例中的 `user`.`log_{north|south|east}` 表至下游TiDB中的 `user`.`log_{north|south|east}` 表。
```
Comment (lilin90) [inline tools/dm/shard-merge-scenario.md]:
```suggestion
2. 合并三个实例中的 `user`.`log_{north|south|east}` 表至下游 TiDB 中的 `user`.`log_{north|south|east}` 表。
```

Comment (lilin90) [reply tools/dm/shard-merge-scenario.md]: Also check all similar expressions and make sure to leave a space between English and Chinese characters.

**Comment by GregoryIan** [comment]
LGTM

File: tools/dm/shard-merge-scenario.md [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+3. 合并三个实例中的 `store_{01|02}`.`sale_{01|02}` 表至下游TiDB中的 `store`.`sale` 表。
+4. 过滤掉三个实例的 `user`.`log_{north|south|east}` 表的所有删除操作。
+5. 过滤掉三个实例的 `user`.`information` 表的所有删除操作。
+6. 过滤掉三个实例的 `store_{01|02}`.`sale_{01|02}` 表的所有删除操作
```
Comment (lilin90) [inline tools/dm/shard-merge-scenario.md]:
```suggestion
6. 过滤掉三个实例的 `store_{01|02}`.`sale_{01|02}` 表的所有删除操作。
```

File: tools/dm/shard-merge-scenario.md [hunk 11]
```diff
@@ -0,0 +1,272 @@
...
+假设同步后下游库结构如下：
+
+| Schema | Tables |
+|:------|:------|
+| user | information, log_north, log_east, log_south|
+| store | sale |
+
+## 同步方案
+
+- 要满足同步需求 #1 和 #2, 配置 [Table routing 规则](/tools/dm/data-synchronization-features.md#table-routing) 如下：
```
Comment (lilin90) [inline tools/dm/shard-merge-scenario.md]:
```suggestion
- 要满足同步需求 #1 和 #2，配置 [Table routing 规则](/tools/dm/data-synchronization-features.md#table-routing) 如下：
```

File: tools/dm/shard-merge-scenario.md [hunk 12]
```diff
@@ -0,0 +1,272 @@
...
+
+    ```yaml
+    routes:
+      ...
+      user-route-rule:
+        schema-pattern: "user"
+        target-schema: "user"
+    ```
+
+- 要满足同步需求 #3, 配置 [Table routing 规则](/tools/dm/data-synchronization-features.md#table-routing) 如下：
```
Comment (lilin90) [inline tools/dm/shard-merge-scenario.md]:
```suggestion
- 要满足同步需求 #3，配置 [Table routing 规则](/tools/dm/data-synchronization-features.md#table-routing) 如下：
```

File: tools/dm/shard-merge-scenario.md [hunk 13]
```diff
@@ -0,0 +1,272 @@
...
+        schema-pattern: "store_*"
+        target-schema: "store"
+      sale-route-rule:
+        schema-pattern: "store_*"
+        table-pattern: "sale_*"
+        target-schema: "store"
+        target-table:  "sale"
+    ```
+
+- 要满足同步需求 #4 和 #5, 配置 [Binlog event filtering 规则](/tools/dm/data-synchronization-features.md#binlog-event-filtering) 如下：
```
Comment (lilin90) [inline tools/dm/shard-merge-scenario.md]:
```suggestion
- 要满足同步需求 #4 和 #5，配置 [Binlog event filtering 规则](/tools/dm/data-synchronization-features.md#binlog-event-filtering) 如下：
```

File: tools/dm/shard-merge-scenario.md [hunk 14]
```diff
@@ -0,0 +1,272 @@
...
+      ...
+      user-filter-rule:
+        schema-pattern: "user"
+        events: ["truncate table", "drop table", "delete"， "drop database"]
+        action: Ignore
+    ```
+
+    > **注意：** 同步需求 #4、#5 和 #7 的操作意味着过滤掉所有对 `user` 库的删除操作，所以此处配置了库级别的过滤规则。但是 `user` 库以后加入表的删除操作也都会被过滤。
+
+- 要满足同步需求 #6, 配置 [Binlog event filtering 规则](/tools/dm/data-synchronization-features.md#binlog-event-filtering) 如下：
```
Comment (lilin90) [inline tools/dm/shard-merge-scenario.md]:
```suggestion
- 要满足同步需求 #6，配置 [Binlog event filtering 规则](/tools/dm/data-synchronization-features.md#binlog-event-filtering) 如下：
```

File: tools/dm/shard-merge-scenario.md [hunk 15]
```diff
@@ -0,0 +1,272 @@
...
+
+    ```yaml
+    black-white-list:
+      log-bak-ignored:
+        ignore-tales:
+        - db-name: "user"
+          tbl-name: "log_bak"
+    ```
+
+- 要满足同步需求 #8, 配置 [Column mapping 规则](/tools/dm/data-synchronization-features.md#column-mapping) 如下：
```
Comment (lilin90) [inline tools/dm/shard-merge-scenario.md]:
```suggestion
- 要满足同步需求 #8，配置 [Column mapping 规则](/tools/dm/data-synchronization-features.md#column-mapping) 如下：
```

File: tools/dm/shard-merge-scenario.md [hunk 16]
```diff
@@ -0,0 +1,272 @@
...
+    ```yaml
+    filters:
+      ...
+      user-filter-rule:
+        schema-pattern: "user"
+        events: ["truncate table", "drop table", "delete"， "drop database"]
+        action: Ignore
+    ```
+
+    > **注意：** 同步需求 #4、#5 和 #7 的操作意味着过滤掉所有对 `user` 库的删除操作，所以此处配置了库级别的过滤规则。但是 `user` 库以后加入表的删除操作也都会被过滤。
```
Comment (lilin90) [inline tools/dm/shard-merge-scenario.md]:
```suggestion
    > **注意**：同步需求 #4、#5 和 #7 的操作意味着过滤掉所有对 `user` 库的删除操作，所以此处配置了库级别的过滤规则。但是 `user` 库以后加入表的删除操作也都会被过滤。
```

**Pull request merged**